sbi/inference/snl/snl.py

```diff
@@ -94,36 +92,23 @@ def __call__(
         Returns:
             Posterior $p(\theta|x_o)$ that can be sampled and evaluated
         """
-        round_description = ""
-        tbar = tqdm(range(num_rounds))
-        for round_ in tbar:
+        num_sims_per_round = self._ensure_list(num_simulations_per_round, num_rounds)
 
-            tbar.set_description(round_description)
+        tbar = tqdm(enumerate(num_sims_per_round))
+        for round_, num_sims in tbar:
 
             # Generate parameters theta from prior in first round, and from most recent
             # posterior estimate in subsequent rounds.
             if round_ == 0:
-                theta, x = simulators.simulate_in_batches(
-                    simulator=self._simulator,
-                    parameter_sample_fn=lambda num_samples: self._prior.sample(
-                        (num_samples,)
-                    ),
-                    num_samples=num_simulations_per_round,
-                    simulation_batch_size=self._simulation_batch_size,
-                )
+                theta = self._prior.sample((num_sims,))
             else:
-                theta, x = simulators.simulate_in_batches(
-                    simulator=self._simulator,
-                    parameter_sample_fn=lambda num_samples: self._neural_posterior.sample(
-                        num_samples
-                    ),
-                    num_samples=num_simulations_per_round,
-                    simulation_batch_size=self._simulation_batch_size,
-                )
+                theta = self._neural_posterior.sample(num_sims)
+
+            x = self._batched_simulator(theta, num_sims)
```

Comment:
should be `self._batched_simulator(theta, self._simulation_batch_size)` no?

Reply:
This is a really, really good catch. Thanks!

Reply:
In fact, this is what was already done by default - `simulation_batch_size` was built-in into the `batched_simulator` from its creation in `NeuralInference` and thus the second argument was unnecessary - the refactor was simply not complete. Fri night...